allay/plugin.py

```diff
@@ -0,0 +1,29 @@
+from beet import Context
+from beet.core.container import Drop
+from beet.core.file import TextFile
+from beet.contrib.messages import Message
+from beet.library.base import NamespaceFile
+from beet.library.data_pack import DataPack
+
+from tokenstream import InvalidSyntax
+
+from functools import cached_property
+
+from .parser import Parser
+
+class AllayMessage(TextFile, NamespaceFile):
+    scope = ("messages",)
+    extension = ".allay"
+
+    @cached_property
+    def parser(self) -> Parser:
+        return Parser()
+
+    def bind(self, pack: DataPack, path: str):
+        json = self.parser.parse(self.text)
```

Comment:
`self.text` is undefined.

Reply:
um, my code ran with this. This might be how `bind` works in `beet`, `self.text` is available through `beet`. Also, this adds `beet` as a dependency. Maybe we should try to keep `beet` as an optional dep though

Reply:
resolved in latest commit